src/broker/broker_config.c

```diff
@@ -1093,6 +1173,87 @@ broker_config_read_internal (const char *conf_file, T_BROKER_INFO * br_info, int
   return -1;
 }
 
+static void
+write_conf_cache (char *broker_conf_file, bool * acl_flag, int *num_broker, int *br_shm_id, char *admin_logfile,
+		  T_BROKER_INFO * br_info, time_t br_conf_mtime)
+{
+  if (broker_conf_file == NULL || num_broker == NULL || br_info == NULL || br_shm_id == NULL)
+    {
+      return;
+    }
+
+  for (int i = 0; i < MAX_NUM_BROKER_FILES; i++)
+    {
+      if (br_conf_info[i].conf_file == NULL)
+	{
+	  br_conf_info[i].conf_file = strdup(broker_conf_file);
+	  br_conf_info[i].num_broker = *num_broker;
+	  br_conf_info[i].br_shm_id = *br_shm_id;
+	  br_conf_info[i].last_modified = br_conf_mtime;
+
+	  if (acl_flag)
+	    {
+	      br_conf_info[i].acl_flag = *acl_flag;
+	    }
+
+	  if (admin_logfile)
+	    {
+	      br_conf_info[i].admin_log_file = strdup (admin_logfile);
+	    }
+
+	  memcpy (br_conf_info[i].br_info, br_info, sizeof (T_BROKER_INFO) * MAX_BROKER_NUM);
+	}
+    }
+}
+
+static void
+read_conf_cache (int cid, bool *acl_flag, int *num_broker, int *br_shm_id, char *logfile, T_BROKER_INFO *br_info)
+{
+  if (cid < 0 || cid > MAX_NUM_BROKER_FILES || br_shm_id == NULL || br_info == NULL)
+    {
+      return;
+    }
+
+  if (acl_flag)
+    {
+      *acl_flag = br_conf_info[cid].acl_flag;
+    }
+
+  if (logfile && br_conf_info[cid].admin_log_file)
+    {
+      strcpy (logfile, br_conf_info[cid].admin_log_file);
+    }
+
+  *num_broker = br_conf_info[cid].num_broker;
+  *br_shm_id = br_conf_info[cid].br_shm_id;
+  memcpy (br_info, br_conf_info[cid].br_info, sizeof (T_BROKER_INFO) * MAX_BROKER_NUM);
+
+  return;
+}
+
+static void
+clear_conf_cache_entry (int cid)
+{
+  if (cid < 0 || cid > MAX_NUM_BROKER_FILES)
+    {
+      return;
+    }
+
+  if (br_conf_info[cid].conf_file)
+    {
+      free (br_conf_info[cid].conf_file);
```

Comment:
I think br_conf_info[cid].conf_file = NULL should be added.

Reply:
I think the following memset will do it, isn't it? It will set 0 to br_conf_info[cid].conf_file also.

`memset (&br_conf_info[cid], 0, sizeof (T_CONF_INFO));`

Reply:
I did not see the memset below. I'm sorry.